serdes-native/src/main/java/io/ballerina/stdlib/serdes/Deserializer.java

```diff
@@ -0,0 +1,345 @@
+                } else {
+                    bArray.append(element);
+                }
+            }
+            return bArray;
+        }
+    }
+
+    private static Map<String, Object> getBallerinaRecordValueFromMessage(DynamicMessage dynamicMessage, Type type,
+                                                                          Descriptor schema) {
+        Map<String, Object> map = new HashMap();
+
+        for (Map.Entry<FieldDescriptor, Object> entry : dynamicMessage.getAllFields().entrySet()) {
+            String fieldName = entry.getKey().getName();
+            Object value = entry.getValue();
+            if (value instanceof DynamicMessage) {
+                DynamicMessage nestedDynamicMessage = (DynamicMessage) value;
+                String fieldType = nestedDynamicMessage.getDescriptorForType().getName();
+                String[] processFieldName = fieldType.split("_");
+                String unionCheck = processFieldName[processFieldName.length - 1];
+                if (unionCheck.contains(UNION_TYPE_IDENTIFIER)) {
+                    Descriptor unionSchema = schema.findNestedTypeByName(fieldType);
+                    Type unionType = null;
+                    RecordType recordType = (RecordType) type;
+                    for (Map.Entry<String, Field> member: recordType.getFields().entrySet()) {
+                        if (member.getKey().equals(fieldName)) {
+                            unionType = member.getValue().getFieldType();
+                            break;
+                        }
+                    }
+                    map.put(fieldName, getBallerinaUnionTypeValueFromMessage(
+                            nestedDynamicMessage,
+                            unionType,
+                            unionSchema
+                    ));
+                } else {
+                    Map<String, Object> nestedMap = getBallerinaRecordValueFromMessage(
+                            nestedDynamicMessage,
+                            type,
+                            schema
+                    );
+                    String recordTypeName = getRecordTypeName(type, fieldName);
+                    BMap<BString, Object> nestedRecord = ValueCreator.createRecordValue(
+                            type.getPackage(),
+                            recordTypeName,
+                            nestedMap
+                    );
+                    map.put(fieldName, nestedRecord);
+                }
+            } else if (value.getClass().getSimpleName().equals(BYTE) || entry.getKey().isRepeated()) {
+                if (!value.getClass().getSimpleName().equals(BYTE)) {
+                    Type elementType = getArrayElementType(type, fieldName);
+                    Object handleArray = getBallerinaArrayValueFromMessage(
+                            value,
+                            elementType,
+                            schema,
+                            1
+                    );
```

Comment:
Can't we put these in a single line? Check for the other similar cases.

Reply:
Updated